src/clojure/puppetlabs/ssl_utils/simple.clj

```diff
@@ -0,0 +1,108 @@
+(ns puppetlabs.ssl-utils.simple
+  (:import (java.util Date))
+  (:require [puppetlabs.ssl-utils.core :as ssl-utils]
+            [clj-time.core :as time]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Predicates
+
+(defn validity-date-range?
+  "Returns true if the given map contains all the fields required to define a
+  validity date range for a certificate."
+  [x]
+  (and (map? x)
+    (instance? Date (:not-before x))
+    (instance? Date (:not-after x))))
+
+(defn ssl-keys?
+  "Returns true if the given map contains all the fields required to define an
+  SSL keypair and associated cetname info."
+  [x]
+  (and (map? x)
+    (ssl-utils/public-key? (:public-key x))
+    (ssl-utils/private-key? (:private-key x))
+    (ssl-utils/valid-x500-name? (:x500-name x))
+    (string? (:certname x))))
+
+(defn ssl-cert?
+  "Returns true if the given map contains all the fields required to define a
+  certificate, keypair, and associated certname info."
+  [x]
+  (and (ssl-keys? x)
+    (ssl-utils/certificate? (:cert x))))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Public
+
+(defn cert-validity-dates
+  "Calculate the not-before & not-after dates that define a certificate's
+   period of validity. The value of `ca-ttl` is expected to be in seconds,
+   and the dates will be based on the current time. Returns a map in the
+   form {:not-before Date :not-after Date}."
+  [ca-ttl]
+  {:pre [(integer? ca-ttl)]
+   :post [(validity-date-range? %)]}
+  (let [now        (time/now)
+        not-before (time/minus now (time/days 1))
+        not-after  (time/plus now (time/secs ca-ttl))]
+    {:not-before (.toDate not-before)
+     :not-after  (.toDate not-after)}))
+
+(def key-length 4096)
```

Comment:
FWIW I usually scale this down to 512 during development since it cuts down the key operation time significantly.  The API currently doesn't allow for any size other than 4096, and you mention this namespace is useful for development...so just something to consider making more flexible.

You might be able to get away with a second-arity of `gen-keys` that takes the keylength.